pyseir/deployment/demographic_mapper.py

```diff
@@ -0,0 +1,729 @@
+                               infection state (susceptible, infected,
+                               hospitalized, etc.) by demographic group
+                               simulated by MLE model. Each time series is
+                               recorded as pd.DataFrame, with dates of
+                               prediction as index.
+              name of compartment include: S - susceptible, E - exposed,
+              A - asymptomatic, I - symptomatic, HGen - in non-ICU, HICU - in
+              ICU, HVent - on ventilator, N - entire population.
+            - <measure>:
+              - <measure_unit>: time series of covid measures by demographic
+                                group predicted using the MLE model.
+                                The time series is recorded as pd.DataFrame,
+                                with dates of prediction as index.
+
+        Returns
+        -------
+          : dict
+            Contains:
+            - compartments:
+              - <compartment>: time series of population at a specific
+                               infection state (susceptible, infected,
+                               hospitalized, etc.) simulated by MLE model
+                               assuming the population has the demographic
+                               distribution of the target population. Each time
+                               series is recorded as pd.DataFrame, with dates of
+                               prediction as index.
+              name of compartment include: S - susceptible, E - exposed,
+              A - asymptomatic, I - symptomatic, HGen - in non-ICU, HICU - in ICU,
+              HVent - on ventilator, N - entire population.
+            - <measure>:
+              - <measure_unit>: time series of covid measures predicted using
+                                the MLE model and averaged over target
+                                demographic distribution (adjusted by risk
+                                modification if relative risk is specified).
+                                The time series is recorded as pd.DataFrame,
+                                with dates of prediction as index.
+        """
+        # calculate weights
+        age_bin_centers = [np.mean(tup) for tup in self.parameters['age_groups']]
+        weights = self.target_age_distribution_pdf(age_bin_centers)
+        if (weights != 1).sum() == 0:
+            logging.warning('no target age distribution is given, measure is aggregated assuming age '
+                            'distrubtion at given FIPS code')
+
+        weights /= weights.sum()
+        demographic_group_size_ratio = weights / (self.parameters['N'] / self.parameters['N'].sum())
+
+        mapped_predictions = defaultdict(dict)
+
+        for c in predictions['compartments']:
+            mapped_predictions['compartments'][c] = predictions['compartments'][c].dot(demographic_group_size_ratio)
```

Comment:
Sorry if I'm missing context, but why are you using `dot` rather than `*` (elementwise multiplication)? What's the reason to return a scalar to store in `mapped_predictions['compartments'][c]`

Reply:
The dot is to calculate average of compartment size weighted by group size ratio. Potentially we could also store the adjusted compartment size by age group (elementwise multiplication), but not sure if that would be necessary since we will just calculate the index based on the aggregated measure.

Reply:
:+1: makes sense